src/ncdiff/netconf.py

```diff
@@ -983,18 +984,31 @@ def node_sub(self, node_self, node_other, depth=0):
                     e = etree.SubElement(
                         child_other, key, nsmap=key_node.nsmap)
                     e.text = key_node.text
+            if s_node.getparent().get('type') == 'case':
+                # key: choice node, value: case node
+                choice_nodes[s_node.getparent().getparent()] = s_node.getparent()
         for child_other in in_o_not_in_s:
             child_self = etree.Element(child_other.tag,
                                        {operation_tag: self.preferred_delete},
                                        nsmap=child_other.nsmap)
             if self.preferred_create != 'merge':
                 child_other.set(operation_tag, self.preferred_create)
             siblings = list(node_self.iterchildren(tag=child_other.tag))
+            s_node = self.device.get_schema_node(child_other)
             if siblings:
                 siblings[-1].addnext(child_self)
             else:
-                node_self.append(child_self)
-            s_node = self.device.get_schema_node(child_other)
+                # Append node if:
+                # Node not in case
+                # Node in case but choice node not in self
+                # Node in case and choice not in self but case also in self
```

Comment:
How about `Node in case and choice node in self and the same case also in self`?

Reply:
I did a unit test for this
https://github.com/CiscoTestAutomation/ncdiff/pull/27/files#diff-a32bac34fbf81e957a4ced64cf60128f21ff9ea8d403fcc89b3b39906a0af8c0R1139

Reply:
You are right. No change is needed.